src/main/java/com/blibli/future/service/FileReaderComponentImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.blibli.future.service;
+
+import com.blibli.future.service.api.FileReaderComponent;
+import org.apache.poi.hssf.usermodel.HSSFWorkbook;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class FileReaderComponentImpl implements FileReaderComponent {
+    @Override
+    public List<String> readFileAsStrings(MultipartFile file) {//Question: Is it Okay to do this? A class just for parsing a file?
+        String filename = file.getContentType();
+
+        if(filename.endsWith(".csv")) {
+            return readCsvAsList(file);
+        } else if(filename.endsWith(".xlsx")){
+            return readXlsxAsList(file);
+        } else if(filename.endsWith(".xls")) {
+            return readXlsAsList(file);
+        } else {
+            return null; //or throw "Unsupported file"
+        }
+    }
+
+    private List<String> readCsvAsList(MultipartFile csvFile) {
+        List<String> result = new ArrayList<>();
+
+        try (InputStream inputStream = csvFile.getInputStream()) {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+            String line;
+            while((line= reader.readLine()) != null) {
+                result.add(line);
+            }
+        } catch (IOException e) {
```

Comment:
Catching the exception here is potentially dangerous and could produce empty result if the file corrupt or something. Maybe `return null` or throw exception that will be handled here https://github.com/blibli-future/athens/pull/1/files#diff-e47c49c4547e3aa56d47d1d685a08224R180